Fw/Python/src/fprime/fbuild/cmake.py

```diff
@@ -338,19 +341,26 @@ def _read_values_from_cache(self, keys, build_dir):
     def _read_cache(self, build_dir):
         """
         Reads the cache from the associated build_dir. This will return a dictionary of cache variable name to
-        its value. This will not update internal state.
+        its value. The first run will call cmake to extract the cache and cache it internally. Subsequent
+        calls will return the cached cache.
 
         :param build_dir: build directory to harvest for cache variables
         :return: {<cmake cache variable>: <cmake cache value>}
         """
+        if self._cmake_cache is not None:
```

Comment:
I wanted to avoid the cost of making multiple calls to `cmake -LA` which is pretty expensive. I'm not sure how caching this could cause any problems, the only time the cache changes is after running generate. The only place the cache is currently used is get the cmake module name, which is only used before a build and when determining targets